### PR TITLE
Drop support for PHP version less than 7.4

### LINKED ISSUE #226: Drop support for unsupported PHP versions
Support for versions less than PHP 7.4 should be dropped so new language features can be used such as typed properties.

[Supported Versions](https://www.php.net/supported-versions.php)

### PR DESCRIPTION
**Pull request by JimTools**
Drop support for PHP version less than 7.4

closes #226

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,19 +20,19 @@
...
-        "psr/log": "^1.0|^2.0|^3.0",
-        "firebase/php-jwt": "^3.0|^4.0|^5.0",
+        "php": "^7.4|^8.0",
+        "psr/log": "^1.1|^2.0|^3.0",
+        "firebase/php-jwt": "^5.0",
         "psr/http-message": "^1.0",
-        "tuupola/http-factory": "^0.4.0|^1.0.2",
-        "tuupola/callable-handler": "^0.3.0|^0.4.0|^1.0",
+        "tuupola/http-factory": "^1.4",
+        "tuupola/callable-handler": "^1.1",
```
Comment (tuupola) [inline composer.json]: Is there a reason to bump the minimum version of dependencies such as this one? Going for latest is ok for dev dependencies but tends to lead to dependency hell in normal dependencies.

Comment (JimTools) [reply composer.json]: @tuupola my main reason behind this was the newer version supports PHP 8, so it would be less of a hassle to upgrade in the future.

**Comment by tuupola** [comment]
Now there is too many things happening in one PR. I already did a separate update for PHPStan and couple of smaller changes. Could you either rebase or do another PR which has only the typed properties and return types. Dependencies only need to be updated if any of the tests fail.

**Comment by tuupola** [comment]
Better late than never. Thanks!

**Pull request merged**